### ISSUE TITLE
Sharpness-Aware Minimization for Efficiently Improving Generalization

### ISSUE THREAD

**Issue by SeanPresent** [iRAIL-SNU/irail-paper-review#9]
### Review

Overparameterization has been noticed in studies for models to achieve better performance. 
There are multiple ways to handle overfitting and generalization issues as data enhancement, regularization, and loss function. And the author claims that loss at a minimum point can be found through better gradient optimization in handling the generalization of the model.

![Screen Shot 2023-02-16 at 10 07 12 AM](https://user-images.githubusercontent.com/90368764/219232274-23781708-980e-428e-862e-6caeda0342f3.png)

Depth models often cause many minimum points of their loss function because of their numerous parameters, and Gradient descent is a suitable approach to finding one of the many minimum points. There is some theoretical analysis at present, it is a flat minimum point is better than a sharp minimal point in general knowledge. Using the idea of sharp minima is much more sensitive for parameters, and even more for input. It is driven by the idea that flat minima are much more stable. Therefore, when optimizing the model, we prefer parameters to be optimized to flat minima rather than sharp minima.

![Screen Shot 2023-02-16 at 10 18 25 AM](https://user-images.githubusercontent.com/90368764/219238093-c8b519ed-8400-49eb-ba5f-d03ab2bcbc08.png)

To improve the generalization of the model, minimizing loss value and loss sharpness, the author introduced SAM’s efficacy. The idea of per-data-point sharpness yielded by m-sharpness could potentially deploy in Mixup for the robust model and semi-supervised learning.

# SHARPNESS-AWARE MINIMIZATION (SAM)

### Paper URL

https://arxiv.org/pdf/2010.01412v3.pdf
